lib/queryBuilder/graphUpserter/UpsertNode.js

```diff
@@ -94,7 +93,7 @@ function hasChanges(currentModel, upsertModel, queryProps) {
   for (let i = 0, l = keys.length; i < l; ++i) {
     const key = keys[i];
 
-    if (!relations[key] && currentModel[key] !== upsertModel[key]) {
```

Comment:
I already accidentally wrote this comment in the commit in your fork :D

Doesn't this cause an update query to be made each time a model has any relations because the equality of two objects (or arrays) is always false? This causes a bunch of useless update queries.

Reply:
Hehe, and I replied in the conversation :)

True, how about:
`if (relations[key] || currentModel[key] !== upsertModel[key])`

Reply:
Last comment was a bit naive of me. I'll look into to it more closely tomorrow.